### PR TITLE
[AC-5622] Add API calls for Criterion and CriterionOptionSpec

### PR DESCRIPTION
**Pull request by jonkiparsky**
* API calls exist for list and detail, including patch and post
* calls work as expected (detail view includes PATCH and list view include POST, as with User)
* attached Postman collection includes all calls (may need to be tweaked)
[localhost.postman_collection.json.txt](https://github.com/masschallenge/impact-api/files/2122133/localhost.postman_collection.json.txt)

* decided to use mixins to reduce code duplication and avoid tangled inheritance messes. we may need to develop standards around sane usage of mixin classes to avoid future design issues. (either as part of this ticket, or on a new ticket)

### CODE REVIEW

**Comment by mc-nathan** [comment]
Functional issue:
It seems wrong to me that according to the OPTIONS, the POST for a Criterion (in the list view) does not expect a judging round id:
```
        "POST": {
            "type": "object",
            "properties": {
                "name": {
                    "type": "string",
                    "required": true
                },
                "type": {
                    "type": "string",
                    "required": true
                }
            }
        }
```

**Comment by mc-nathan** [comment]
`judging_round_id` is also missing from the OPTIONS for the PATCH on a specific Criterion.

**Comment by mc-nathan** [comment]
Functional issue: The `id` should never get patched, but it's in the OPTIONS for the PATCH.

**Comment by mc-nathan** [comment]
Looks like this is strictly an issue with the OPTIONS since the actual PATCH call does take `judging_round_id` and rejects calls that specify `id`.

**Comment by mc-nathan** [comment]
Functional issue: `count`, `weight` and `criterion_id` are missing from the OPTIONs for the CriterionOptionSpec POST.
```
        "POST": {
            "type": "object",
            "properties": {
                "option": {
                    "type": "string",
                    "required": true
                }
            }
        }
```

**Comment by mc-nathan** [comment]
Similarly for PATCH, the `count`, `weight` and `criterion_id` are missing and `id` is listed.

**Review by mc-nathan — CHANGES_REQUESTED** [review]
Some minor code-ish stuff, but some important functional issues.

File: web/impact/impact/v1/helpers/user_helper.py [hunk 1]
```diff
@@ -133,15 +133,15 @@ def save(self):
...
-        keys += ProfileHelper.CORE_KEYS
-    else:
-        keys += ProfileHelper.CORE_PATCH_KEYS
-    if not user_type or user_type == ExpertProfile.user_type:
-        keys += ProfileHelper.EXPERT_KEYS
-    if not user_type or user_type == EntrepreneurProfile.user_type:
-        keys += ProfileHelper.ENTREPRENEUR_KEYS
-    return VALID_KEYS_NOTE.format(keys)
+    @classmethod
+    def valid_keys(cls, user_type, post=False):
```
[the remark this answers is not in the document]
Comment (jonkiparsky) [reply web/impact/impact/v1/helpers/user_helper.py]: okay

**Review by mc-nathan — APPROVED** [review]

**Pull request merged**